### PR TITLE
Add Wisun extended key usage

### PR DESCRIPTION
**Pull request by RonEld**
## Description
Add the Wisun Fan device extended key usage, and tests for extended key usage oid.

## Status
**READY**

## Requires Backporting
NO  

## Todos
- [x] Tests
- [ ] Documentation
- [x] Changelog updated
- [ ] Backported

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -5,6 +5,7 @@ mbed TLS ChangeLog (Sorted per branch, date)
 Features
    * Add the Any Policy certificate policy oid, as defined in
      rfc 5280 section 4.2.1.4.
+   * Add the Wisun Fan device extended key usage.
```
Comment (Patater) [inline ChangeLog]: This should use proper capitalization: `Wi-SUN`

File: include/mbedtls/oid.h [hunk 2]
```diff
@@ -206,6 +207,13 @@
 #define MBEDTLS_OID_TIME_STAMPING               MBEDTLS_OID_KP "\x08" /**< id-kp-timeStamping OBJECT IDENTIFIER ::= { id-kp 8 } */
 #define MBEDTLS_OID_OCSP_SIGNING                MBEDTLS_OID_KP "\x09" /**< id-kp-OCSPSigning OBJECT IDENTIFIER ::= { id-kp 9 } */
 
+/**
+ * Wisun aliance Field Area Network
```
Comment (Patater) [inline include/mbedtls/oid.h]: Use correct case: `Wi-SUN`

File: library/oid.c [hunk 3]
```diff
@@ -284,12 +284,13 @@ FN_OID_GET_ATTR1(mbedtls_oid_get_x509_ext_type, oid_x509_ext_t, x509_ext, int, e
...
-    { ADD_LEN( MBEDTLS_OID_EMAIL_PROTECTION ), "id-kp-emailProtection", "E-mail Protection" },
-    { ADD_LEN( MBEDTLS_OID_TIME_STAMPING ),    "id-kp-timeStamping",    "Time Stamping" },
-    { ADD_LEN( MBEDTLS_OID_OCSP_SIGNING ),     "id-kp-OCSPSigning",     "OCSP Signing" },
+    { ADD_LEN( MBEDTLS_OID_SERVER_AUTH ),      "id-kp-serverAuth",       "TLS Web Server Authentication" },
+    { ADD_LEN( MBEDTLS_OID_CLIENT_AUTH ),      "id-kp-clientAuth",       "TLS Web Client Authentication" },
+    { ADD_LEN( MBEDTLS_OID_CODE_SIGNING ),     "id-kp-codeSigning",      "Code Signing" },
+    { ADD_LEN( MBEDTLS_OID_EMAIL_PROTECTION ), "id-kp-emailProtection",  "E-mail Protection" },
+    { ADD_LEN( MBEDTLS_OID_TIME_STAMPING ),    "id-kp-timeStamping",     "Time Stamping" },
+    { ADD_LEN( MBEDTLS_OID_OCSP_SIGNING ),     "id-kp-OCSPSigning",      "OCSP Signing" },
+    { ADD_LEN( MBEDTLS_OID_WISUN_FAN ),        "id-kp-wisun-fan-device", "Wi-Sun alliance field area network" },
```
Comment (Patater) [inline library/oid.c]: Use correct case: `Wi-SUN Alliance`

File: tests/suites/test_suite_oid.function [hunk 4]
```diff
@@ -32,3 +32,26 @@ void oid_get_certificate_policies( data_t * oid, char * result_str )
...
+    asn1_buf.len = oid->len;
+
+    ret = mbedtls_oid_get_extended_key_usage( &asn1_buf, &desc );
+    if( strlen( result_str ) == 0 )
+    {
+        TEST_ASSERT( ret == MBEDTLS_ERR_OID_NOT_FOUND );
+    }
+    else
+    {
+        TEST_ASSERT( strcmp( ( char* )desc, result_str ) == 0 );
```
Comment (Patater) [inline tests/suites/test_suite_oid.function]: style

```diff
-        TEST_ASSERT( strcmp( ( char* )desc, result_str ) == 0 );
+        TEST_ASSERT( strcmp( ( char * )desc, result_str ) == 0 );
```

File: tests/suites/test_suite_oid.function [hunk 5]
```diff
@@ -32,3 +32,26 @@ void oid_get_certificate_policies( data_t * oid, char * result_str )
     }
 }
 /* END_CASE */
+
+/* BEGIN_CASE */
+void oid_get_extended_key_usage( data_t * oid, char * result_str )
```
Comment (Patater) [inline tests/suites/test_suite_oid.function]: style: we tend to put the `*` next to the variable, without spaces between it and the variable name.

Comment (RonEld) [reply tests/suites/test_suite_oid.function]: It did look strange to me, but I took the example from [here](https://github.com/ARMmbed/mbedtls/blob/development/tests/suites/test_suite_x509parse.function#L166), so I thought that perhaps the tests are exception

**Comment by RonEld** [comment]
@Patater Thank you for your comments.
I amended the commit with your comments, as these were small changes.
please review

File: ChangeLog [hunk 6]
```diff
@@ -5,6 +5,7 @@ mbed TLS ChangeLog (Sorted per branch, date)
 Features
    * Add the Any Policy certificate policy oid, as defined in
      rfc 5280 section 4.2.1.4.
+   * Add the Wi-SUN Fan device extended key usage.
```
Comment (Patater) [inline ChangeLog]: This should be "FAN", right?

File: include/mbedtls/oid.h [hunk 7]
```diff
@@ -206,6 +207,13 @@
 #define MBEDTLS_OID_TIME_STAMPING               MBEDTLS_OID_KP "\x08" /**< id-kp-timeStamping OBJECT IDENTIFIER ::= { id-kp 8 } */
 #define MBEDTLS_OID_OCSP_SIGNING                MBEDTLS_OID_KP "\x09" /**< id-kp-OCSPSigning OBJECT IDENTIFIER ::= { id-kp 9 } */
 
+/**
+ * Wi-SUN aliance Field Area Network
```
Comment (Patater) [inline include/mbedtls/oid.h]:
```diff
- aliance
+ Alliance
```

File: library/oid.c [hunk 8]
```diff
@@ -284,12 +284,13 @@ FN_OID_GET_ATTR1(mbedtls_oid_get_x509_ext_type, oid_x509_ext_t, x509_ext, int, e
...
-    { ADD_LEN( MBEDTLS_OID_EMAIL_PROTECTION ), "id-kp-emailProtection", "E-mail Protection" },
-    { ADD_LEN( MBEDTLS_OID_TIME_STAMPING ),    "id-kp-timeStamping",    "Time Stamping" },
-    { ADD_LEN( MBEDTLS_OID_OCSP_SIGNING ),     "id-kp-OCSPSigning",     "OCSP Signing" },
+    { ADD_LEN( MBEDTLS_OID_SERVER_AUTH ),      "id-kp-serverAuth",       "TLS Web Server Authentication" },
+    { ADD_LEN( MBEDTLS_OID_CLIENT_AUTH ),      "id-kp-clientAuth",       "TLS Web Client Authentication" },
+    { ADD_LEN( MBEDTLS_OID_CODE_SIGNING ),     "id-kp-codeSigning",      "Code Signing" },
+    { ADD_LEN( MBEDTLS_OID_EMAIL_PROTECTION ), "id-kp-emailProtection",  "E-mail Protection" },
+    { ADD_LEN( MBEDTLS_OID_TIME_STAMPING ),    "id-kp-timeStamping",     "Time Stamping" },
+    { ADD_LEN( MBEDTLS_OID_OCSP_SIGNING ),     "id-kp-OCSPSigning",      "OCSP Signing" },
+    { ADD_LEN( MBEDTLS_OID_WISUN_FAN ),        "id-kp-wisun-fan-device", "Wi-SUN alliance field area network" },
```
Comment (Patater) [inline library/oid.c]: Should we make the following change? At least Alliance should be capitalized, I suspect.

```diff
- Wi-SUN alliance field area network
+ Wi-SUN Alliance field area network (FAN)
```

File: tests/suites/test_suite_oid.function [hunk 9]
```diff
@@ -32,3 +32,26 @@ void oid_get_certificate_policies( data_t * oid, char * result_str )
     }
 }
 /* END_CASE */
+
+/* BEGIN_CASE */
+void oid_get_extended_key_usage( data_t * oid, char *result_str )
```
Comment (Patater) [inline tests/suites/test_suite_oid.function]: Suggested change:
```diff
- void oid_get_extended_key_usage( data_t * oid, char *result_str )
+ void oid_get_extended_key_usage( data_t *oid, char *result_str )
```

**Comment by RonEld** [comment]
@Patater Thanks for your comments.
I amended the commit with your comments

File: ChangeLog [hunk 10]
```diff
@@ -5,6 +5,7 @@ mbed TLS ChangeLog (Sorted per branch, date)
 Features
    * Add the Any Policy certificate policy oid, as defined in
      rfc 5280 section 4.2.1.4.
+   * Add the Wi-SUN Field Area Network(FAN) device extended key usage.
```
Comment (Patater) [inline ChangeLog]: Recommend adding a space after "Network".

Comment (RonEld) [reply ChangeLog]: done

**Pull request merged**